pkg/operator/ceph/cluster/controller.go

```diff
@@ -780,6 +782,53 @@ func (c *ClusterController) onDeviceCMUpdate(oldObj, newObj interface{}) {
 // Delete event functions
 // ************************************************************************************************
 
+func (c *ClusterController) dataDirHostPathCleanUp(clust *cephv1.CephCluster) {
+	logger.Infof("starting data dir host path clean up for cluster - %+v", clust)
+	command := []string{"rm", "rf", clust.Spec.DataDirHostPath}
+	volumes := []v1.Volume{}
+	volumeMounts := []v1.VolumeMount{}
+	validNodes := k8sutil.GetValidNodes(clust.Spec.Storage, c.context.Clientset, cephv1.GetOSDPlacement(clust.Spec.Placement))
```

Comment:
We should try to avoid using `GetValidNodes`, because we can't be sure we are duplicating k8s' scheduling logic properly. Our goal should be to run this on all nodes that have had or could have had mon or OSD pods on them. I think this might mean creating a job to wipe mons and a separate job to wipe OSDs and to apply the appropriate placement items to the jobs as defined in the CRD.